sound/soc/sof/intel/hda-pcm.c

```diff
@@ -15,7 +15,6 @@
  * Hardware interface for generic Intel audio DSP HDA IP
  */
 
-#include <sound/hdaudio_ext.h>
 #include <sound/hda_register.h>
```

Comment:
Actually, hda-pcm.c depends on sound/hdaudio_ext.h. For example it will use stream_to_hdac_ext_stream() which is defined in sound/hdaudio_ext.h.

However it is still working if we remove "#include <sound/hdaudio_ext.h>". This is because we include "hda.h" in hda-pcm.c, and hda.h will include hdaudio_ext.h

So in this case, my question is: should we remove hdaudio_ext.h and let hda-pcm.c depends on "hda.h" including hdaudio_ext.h or let hda-pcm.c include hdaudio_ext.h itself?

Reply:
@libinyang my tests show it's not needed.
What I care about it to have a working compilation when HDA_CORE is not enabled in the Kconfig.

Reply:
I prefer the first option

Reply:
> @libinyang my tests show it's not needed.
> What I care about it to have a working compilation when HDA_CORE is not enabled in the Kconfig.

@plbossart As it will include hda.h and hda.h will include <sound/hdaudio_ext.h>, it means hda-pcm.c will always include <sound/hdaudio_ext.h> implicitly. So the patch should always work. If it doesn't work,  it's not this patch's issue and it means somewhere else is wrong.

If we prefer to remove the <sound/hdaudio_ext.h> as @RanderWang said, we I think we should modify the commit description. Just say something like "hdaudio_ext.h is not necessary" is enough. "Nothing depends on definitions in hdaudio_ext.h" is not precise, as hda-pcm.c does use some definitions in hdaudio_ext.h.

Reply:
which ones?

Reply:
@plbossart  stream_to_hdac_ext_stream(), for example.

Reply:
this is a define, we don't care.
I am talking about actual functions.

Reply:
OK, I believe the patch is OK on different kernel config for the actual functions. The patch is OK to me.